sdks/java/io/snowflake/src/test/java/org/apache/beam/sdk/io/snowflake/test/SnowflakeIOIT.java

```diff
@@ -0,0 +1,174 @@
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Combine;
+import org.apache.beam.sdk.transforms.Count;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.PCollection;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+
+/**
+ * A test of {@link org.apache.beam.sdk.io.snowflake.SnowflakeIO} on an independent Snowflake
+ * instance.
+ *
+ * <p>This test requires a running instance of Snowflake, configured for your GCP account. Pass in
+ * connection information using PipelineOptions:
+ *
+ * <pre>
+ * ./gradlew -p sdks/java/io/snowflake integrationTest -DintegrationTestPipelineOptions='[
+ * "--serverName=<YOUR SNOWFLAKE SERVER NAME>",
+ * "--username=<USERNAME>",
+ * "--password=<PASSWORD>",
+ * "--database=<DATABASE NAME>",
+ * "--schema=<SCHEMA NAME>",
+ * "--stagingBucketName=gs://<GCS BUCKET NAME>",
+ * "--storageIntegrationName=<STORAGE INTEGRATION NAME>",
+ * "--numberOfRecords=<1000, 100000, 600000, 5000000>",
+ * "--runner=DataflowRunner",
+ * "--region=<GCP REGION FOR DATAFLOW RUNNER>",
+ * "--project=<GCP_PROJECT>"]'
+ * --tests org.apache.beam.sdk.io.snowflake.test.SnowflakeIOIT
+ * -DintegrationTestRunner=dataflow
+ * </pre>
+ */
+public class SnowflakeIOIT {
+  private static final String tableName = "IOIT";
+
+  private static SnowflakeIO.DataSourceConfiguration dataSourceConfiguration;
+  private static int numberOfRecords;
+  private static String stagingBucketName;
+  private static String storageIntegrationName;
+
+  @Rule public TestPipeline pipelineWrite = TestPipeline.create();
+  @Rule public TestPipeline pipelineRead = TestPipeline.create();
+
+  @BeforeClass
+  public static void setup() throws SQLException {
+    SnowflakeIOITPipelineOptions options =
+        readIOTestPipelineOptions(SnowflakeIOITPipelineOptions.class);
+
+    numberOfRecords = options.getNumberOfRecords();
+    stagingBucketName = options.getStagingBucketName();
+    storageIntegrationName = options.getStorageIntegrationName();
+
+    dataSourceConfiguration =
+        SnowflakeIO.DataSourceConfiguration.create(SnowflakeCredentialsFactory.of(options))
+            .withDatabase(options.getDatabase())
+            .withRole(options.getRole())
+            .withWarehouse(options.getWarehouse())
+            .withServerName(options.getServerName())
+            .withSchema(options.getSchema());
+
+    TestUtils.runConnectionWithStatement(
+        dataSourceConfiguration.buildDatasource(),
+        String.format(
+            "CREATE OR REPLACE TABLE \"%s\" (\"ID\" INTEGER, \"NAME\" STRING)", tableName));
+  }
+
+  @Test
+  public void testWriteThenRead() {
+    PipelineResult writeResult = runWrite();
+    writeResult.waitUntilFinish();
+
+    PipelineResult readResult = runRead();
+    readResult.waitUntilFinish();
+  }
+
+  @AfterClass
+  @SuppressWarnings("StringSplitter")
```

Comment:
Maybe it's better to cure split instead of Suppress? https://errorprone.info/bugpattern/StringSplitter

Reply:
Actually it's leftover from pre-Filesystems change :D thanks for catching